lib/libzfs/libzfs_import.c

```diff
@@ -1675,6 +1675,45 @@ zpool_clear_label(int fd)
 	return (0);
 }
 
+/*
+ * Scan a list of devices for zfs devices.
+ */
+static int
+zpool_find_import_path(libzfs_handle_t *hdl, kmutex_t *lock,
+    avl_tree_t **slice_cache, char **path, int paths)
+{
+	avl_tree_t *cache;
+	rdsk_node_t *slice;
+	int i, error;
+
+	*slice_cache = NULL;
+	cache = zfs_alloc(hdl, sizeof (avl_tree_t));
+	avl_create(cache, slice_cache_compare, sizeof (rdsk_node_t),
+	    offsetof(rdsk_node_t, rn_node));
+
+	for (i = 0; i < paths; i++) {
+		slice = zfs_alloc(hdl, sizeof (rdsk_node_t));
+		error = asprintf(&slice->rn_name, "%s", path[i]);
```

Comment:
Can you use `zfs_strdup()` here which can't fail since you're only making a copy.

```c
slice->rn_name = zfs_strdup(hdl, path[i]);
```